packages/amesos2/src/Amesos2_Basker_def.hpp

```diff
@@ -179,8 +183,16 @@ Basker<Matrix,Vector>::symbolicFactorization_impl()
         auto sp_colind = this->matrixA_->returnColInd();
           TEUCHOS_TEST_FOR_EXCEPTION(sp_colind == nullptr,
             std::runtime_error, "Amesos2 Runtime Error: sp_colind returned null ");
+#ifndef HAVE_TEUCHOS_COMPLEX
         auto sp_values = this->matrixA_->returnValues();
-          TEUCHOS_TEST_FOR_EXCEPTION(sp_values == nullptr,
+#else
+    // NDE: 09/25/2017
+    // Cannot convert Kokkos::complex<T>* to std::complex<T>*; in this case, use reinterpret_cast
+        using complex_type = typename Util::getStdCplxType< magnitude_type, typename matrix_adapter_type::spmtx_vals_t >::type;
+        complex_type * sp_values = nullptr;
+        sp_values = reinterpret_cast< complex_type * > ( this->matrixA_->returnValues() );
```

Comment:
@mhoemmen if you have a moment could you comment on my use of r`einterpret_cast` here, needed to convert `Kokkos::complex<T>*` to `std::complex<T>*`, specifically is there a better way I should be going about this or any possible bad situations you may foresee?